codegen/src/main/java/org/web3j/codegen/unit/gen/ArgumentResolver.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 Web3 Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.web3j.codegen.unit.gen;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.testcontainers.shaded.org.apache.commons.lang.ArrayUtils;
+
+import org.web3j.tuples.Tuple;
+
+import static org.web3j.codegen.unit.gen.utills.NameUtils.returnTypeAsLiteral;
+import static org.web3j.codegen.unit.gen.utills.NameUtils.toCamelCase;
+
+public class ArgumentResolver {
```

Comment:
Please can we include a comment that describes the intent of all classes at the start?

Reply:
Comments have been added where necessary.